server/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java

```diff
@@ -333,10 +332,12 @@ private static Fields generateTermVectorsFromDoc(IndexShard indexShard, TermVect
     public static String[] getValues(IndexableField[] fields) {
         List<String> result = new ArrayList<>();
         for (IndexableField field : fields) {
-            if (field.fieldType() instanceof KeywordFieldMapper.KeywordFieldType) {
-                result.add(field.binaryValue().utf8ToString());
-            } else if (field.fieldType() instanceof StringFieldType) {
-                result.add(field.stringValue());
+            if (field.fieldType().indexOptions() != IndexOptions.NONE) {
+                if (field.binaryValue() != null) {
+                    result.add(field.binaryValue().utf8ToString());
+                } else {
+                    result.add(field.stringValue());
+                }
```

Comment:
I got lost trying to figure out if the updated if is equivalent to the previous one.

Reply:
Point fields all have IndexOptions.NONE set, so we should only encounter keyword (binary) fields and text fields here.  Text fields return `null` from `field.binaryValue()`, so we check that first to see if it's a keyword field and if so extract the binary value; otherwise it's a text field and so we get the string value.  The tests all seem happy...